src/cursor.c

```diff
@@ -296,6 +296,10 @@ process_cursor_motion(struct server *server, uint32_t time)
 			ly = view->y;
 		} else if (wlr_surface_is_layer_surface(server->seat.pressed.surface)) {
 			wlr_scene_node_coords(server->seat.pressed.node, &lx, &ly);
+#if HAVE_XWAYLAND
+		} else if (server->seat.pressed.is_unmanaged) {
+			wlr_scene_node_coords(server->seat.pressed.node, &lx, &ly);
+#endif
```

Comment:
I think this could be replaced by `if(node->parent == server->unmanaged_tree)` so we don't need the `is_managed` bool in `seat->pressed`.